VDF.Core/ScanEngine.cs

```diff
@@ -214,35 +214,53 @@ public sealed class ScanEngine {
 					while (pauseTokenSource.IsPaused) Thread.Sleep(50);
 
 					var entry = ScanList[i];
+					float[] percent = new float[positionList.Count];
+					Dictionary<double, byte[]?>[] grayBytes = new Dictionary<double, byte[]?>[2];
+					grayBytes[0] = entry.grayBytes;
+					
+					if (Settings.CompareHorizontallyFlipped)
+					{
+						grayBytes[1] = new Dictionary<double, byte[]?>();
+						if (entry.IsImage)
+							grayBytes[1].Add(0, GrayBytesUtils.FlipGrayScale(grayBytes[0][0]!, grayScaleWidth));
+						else
+							for (var j = 0; j < positionList.Count; j++)
+							{
+								var idx = entry.GetGrayBytesIndex(positionList[j]);
+								grayBytes[1].Add(idx, GrayBytesUtils.FlipGrayScale(grayBytes[0][idx]!, grayScaleWidth));
+							}
+					}
+
 					for (var n = i + 1; n < ScanList.Count; n++) {
 						var compItem = ScanList[n];
-						if (entry.IsImage && !compItem.IsImage) continue;
-						var duplicateCounter = 0;
-						float[] percent;
-						if (entry.IsImage) {
-							percent = new float[1];
-							percent[0] = GrayBytesUtils.PercentageDifference(entry.grayBytes[0]!, compItem.grayBytes[0]!);
-							if (percent[0] < percentageDifference)
-								duplicateCounter++;
-						}
-						else {
-							percent = new float[positionList.Count];
-							for (var j = 0; j < positionList.Count; j++) {
-								percent[j] =
-									GrayBytesUtils.PercentageDifference(entry.grayBytes[entry.GetGrayBytesIndex(positionList[j])]!, compItem.grayBytes[compItem.GetGrayBytesIndex(positionList[j])]!);
-								if (percent[j] < percentageDifference) {
-									duplicateCounter++;
+						if (entry.IsImage && !compItem.IsImage) 
+							continue;
+						var isDuplicate = true;
+						int flip = 0;
+						do {
+							if (entry.IsImage) {
+								percent[0] = GrayBytesUtils.PercentageDifference(grayBytes[flip][0]!, compItem.grayBytes[0]!);
+								isDuplicate = (percent[0] < percentageDifference);
+							}
+							else {
+								for (var j = 0; j < positionList.Count; j++) {
+									percent[j] =
+										GrayBytesUtils.PercentageDifference(grayBytes[flip][entry.GetGrayBytesIndex(positionList[j])]!, compItem.grayBytes[compItem.GetGrayBytesIndex(positionList[j])]!);
+									if (!(percent[j] < percentageDifference)) {
+										isDuplicate = false;
+										break; 
+									}
 								}
-								else { break; }
 							}
-						}
-
+						} while(!isDuplicate && Settings.CompareHorizontallyFlipped && flip++ == 0);
```

Comment:
This is also hard to read. It takes a while to understand what the loop does.

Reply:
That was my way of merging a condition :-)

But probably you would think about something like: (?)
```c#
for (flip = 0; flip < Settings.CompareHorizontallyFlipped ? 2 : 1; flip++)
{
 ...
if (isDuplicate)
     break;
}
```

Reply:
Yeah